### PR TITLE
Simplify DirectorySeparatorChar in PathUtilities

### PR DESCRIPTION
**Pull request by Youssef1313**

### CODE REVIEW

File: src/Compilers/Core/Portable/FileSystem/PathUtilities.cs [hunk 1]
```diff
@@ -21,7 +21,7 @@ internal static class PathUtilities
     {
         // We consider '/' a directory separator on Unix like systems. 
         // On Windows both / and \ are equally accepted.
-        internal static readonly char DirectorySeparatorChar = PlatformInformation.IsUnix ? '/' : '\\';
+        internal static readonly char DirectorySeparatorChar = Path.DirectorySeparatorChar;
```
Comment (AlekseyTs) [inline src/Compilers/Core/Portable/FileSystem/PathUtilities.cs:24]: >internal static readonly char DirectorySeparatorChar = Path.DirectorySeparatorChar; [](http://example.com/codeflow?start=8&length=83)

Would it make sense to convert this to a property? It looks like we are storing value of one readonly field in another readonly field.

File: src/Compilers/Core/Portable/FileSystem/PathUtilities.cs [hunk 2]
```diff
@@ -21,7 +21,7 @@ internal static class PathUtilities
     {
         // We consider '/' a directory separator on Unix like systems. 
         // On Windows both / and \ are equally accepted.
-        internal static readonly char DirectorySeparatorChar = PlatformInformation.IsUnix ? '/' : '\\';
+        internal static char DirectorySeparatorChar => Path.DirectorySeparatorChar;
```
Comment (CyrusNajmabadi) [inline src/Compilers/Core/Portable/FileSystem/PathUtilities.cs:24]: why not keep a static-readonly?  once computed, it's not going ot change :)

Comment (Youssef1313) [reply src/Compilers/Core/Portable/FileSystem/PathUtilities.cs:24]: @CyrusNajmabadi This was suggested by Aleksey.

Comment (CyrusNajmabadi) [reply src/Compilers/Core/Portable/FileSystem/PathUtilities.cs:24]: ah, lol.  ok :)

**Review by CyrusNajmabadi — APPROVED** [review]

**Comment by jaredpar** [comment]
The failed GitHub checks are a GitHub bug, not a bug in the change. The GitHub issue was, hopefully, resolved yesterday but PRs didn't update for it. Merging.

**Pull request merged**